airflow/sensors/external_task.py

```diff
@@ -360,6 +362,93 @@ def _handle_execution_date_fn(self, context) -> Any:
         return kwargs_callable(logical_date, **kwargs)
 
 
+class ExternalTaskAsyncSensor(ExternalTaskSensor):
```

Comment:
Why a separate operator ? and not a deferrable mode in the existing one?

Reply:
Hi Team! :)
This sensor was implemented as a separate AsyncSensor to follow our discussion from the PR for big query operators: https://github.com/apache/airflow/pull/26156

So, as Sensors with deferrable option are not sensors, but mostly Operators with execute() method, this Sensor was also implemented as a separate class with deferrable mode.

I will update this PR to fix static checks and reach you later for final review :)